src/client.ts

```diff
@@ -1300,28 +934,24 @@ export class Client {
     const secret = this._getSharedSecret();
     const encData = encRes.slice(0, ENC_MSG_LEN);
     const res = aes256_decrypt(encData, secret);
+    if (!res) {
+      throw new Error('Error handling getAddresses response')
+    }
     // len does not include a 65-byte pubkey that prefies each encResponse
     len += 65;
     // Validate checksum. It will be the last 4 bytes of the decrypted payload. The length of the
     // decrypted payload will be fixed for each given message type.
     const toCheck = res.slice(0, len);
     const cs = parseInt(`0x${res.slice(len, len + 4).toString('hex')}`);
     const csCheck = checksum(toCheck);
-    if (cs !== csCheck)
-      return {
-        err: `Checksum mismatch in response from Lattice (calculated ${csCheck}, wanted ${cs})`,
-      };
+    if (cs !== csCheck) {
+      throw new Error(`Checksum mismatch in response from Lattice (calculated ${csCheck}, wanted ${cs})`)
+    }
 
     // First 65 bytes is the next ephemeral pubkey
     const pub = res.slice(0, 65).toString('hex');
-    try {
-      this.ephemeralPub = getP256KeyPairFromPub(pub);
-      return { err: null, data: res };
-    } catch (e) {
-      return {
-        err: `Error handling getAddresses response: ${e.toString()}`,
-      };
-    }
+    this.ephemeralPub = getP256KeyPairFromPub(pub);
+    return res
```

Comment:
@alex-miller-0 Here's where the return value of `_handleEncResponse` was changed.

Reply:
Oh ty. Missed it in the diff